frontend/src/components/elements/VegaLiteChart/VegaLiteChart.tsx

```diff
@@ -137,6 +147,18 @@ class VegaLiteChart extends React.PureComponent<PropsWithHeight, State> {
     return spec
   }
 
+  /**
+   * Finalize the view so it can be garbage collected. This should be done
+   * when a new view is created, and when the component unmounts.
+   */
+  private finalizeView = (): any => {
+    if (this.vegaFinalizer) {
+      this.vegaFinalizer()
```

Comment:
should we also call `this.vegaView.finalize()` here?

Reply:
No, the finalizer function calls `vegaView.finalize`

Reply:
Or maybe a separate `if` where we check if we have a `this.vegaView` and call finalize in there?